### ISSUE TITLE
[ROOT] YouTube Vanced disappears after rebooting the phone

### ISSUE THREAD

**Issue by dreamwhite** [YTVanced/VancedManager#212]
**Please only report your issue here, if all points below are true**
- I installed the App from [vanced.app](https://vanced.app), this github repository or the Vanced Discord server
- I am using the latest version
- This is an issue in the Vanced Manager app (NOT Youtube Vanced)
- This issue keeps re-occurring every time I try
- For MIUI users: I disabled MIUI optimisation
- For root users: I disabled apk Signature Verification

**Phone Specifications:**
- Brand: OnePlus 6T
- Operating System: crDroid 6.11 
- Android Version: Android 10
- Vanced Manager Version: 1.3.1 (Arcturus)

**Please describe the problem you are having in as much detail as possible:**
I've:

1. Installed Vanced Manager
2. Granted Root Permissions to the app
3. Installed Lucky Patcher
4. Applied "disable APK Verification" patches
5. Installed YouTube Vanced (Dark version) from Vanced Manager (ROOT section)  
6. Rebooted the phone

After following the above steps, YouTube Vanced app disappears and I have to reinstall it.
The version I'm trying to install is 15.40.37 and I can't manage to fix this problem.

Do you have any idea? 

Also, how can I correctly detach the application from Google Play Store? 
I read online about `touch /cache/enable_detach` but I don't know if this is the correct solution.
TUAIA

**Comment by Justdigit** [comment]
I have the exact same problem.
After updating the latest MIUI ROM the problem occurred.

Checked all above steps but it keeps de-installing Vanced.
I am using the root version.

**Comment by Justdigit** [comment]
@dreamwhite did you install latest Xiaomi.eu Rom? Or something similar??

I have reverted back to older one.... And vanced sticks there. Looks like something has changed on xiaomi side.

**Comment by dreamwhite** [comment]
I'm on OnePlus 6T : with crDroid 6.11 :/

**Comment by ruffaloo** [comment]
> @dreamwhite did you install latest Xiaomi.eu Rom? Or something similar??
> 
> I have reverted back to older one.... And vanced sticks there. Looks like something has changed on xiaomi side.

I have the same Problem on my Oneplus 7 pro. I'm running on stock Rom Android Q. How can downgrade YT Vanced [ROOT]? Is it possible in the vanced manager App?

**Comment by Justdigit** [comment]
> > @dreamwhite did you install latest Xiaomi.eu Rom? Or something similar??
> > I have reverted back to older one.... And vanced sticks there. Looks like something has changed on xiaomi side.
> 
> I have the same Problem on my Oneplus 7 pro. I'm running on stock Rom Android Q. How can downgrade YT Vanced [ROOT]? Is it possible in the vanced manager App?

I have flashed an older Rom on my xiaomi didn't do Anything with Vanced manager.

**Comment by Besuz** [comment]
Having the exact same issue, too:

Brand: OnePlus 5T
Operating System: LineageOS 16 (nightly 20201027)
Android Version: Android 9
Vanced Manager Version: 1.3.1
YT Vanced: 15.40.37

**Comment by ruffaloo** [comment]
I guess it's probably an issue with YT Vanced 15.40.37. I will use the non-root version now. Hopefully it'll be fixed soon.

**Comment by X1nto** [comment]
It's not an issue related to vanced, but rather an issue with our new installation method which doesn't require signature verification to be disabled. You might want to use nonroot until (if) this issue gets resolved

**Comment by heysachin** [comment]
Facing the same issue also. Uninstalling the updates to youtube app from app info and reinstalling via vanced manager seems to get YouTube (vanced) working till the next reboot and I have to do the same thing again.

**Comment by dreamwhite** [comment]
## UPDATE: BUG FIXED
It seems that updating:

- Vanced Manager to v1.3.1
- YouTube Vanced to v15.43.32

This issue can be closed

**Comment by heysachin** [comment]
The issue still persists for me even after updating everything to latest version

**Comment by ruffaloo** [comment]
The Issue is not solved for me. After reboot YouTube app keeps crashing.

**Comment by JTandT** [comment]
For me after rebooting the YT app crashes on startup.

Vanced Manager v1.3.1
YouTube Vanced v15.43.32
LG V30 on Android 9

**Comment by CurssedCoffin** [comment]
it still happens with the newest version of manager and vanced app.

**Comment by CurssedCoffin** [comment]
also i can install old versions by zip all the apks in one zip and install with SAI but cannot do the same at 15.43.32.
SAI error at signature inconsistent

**Comment by Justdigit** [comment]
For me the issue is still there

**Comment by GeorgeIvory** [comment]
Same issue. Issue being "youtube vanced uninstalls after each reboot of the device". The last versions of both. 1.3.1 and 15.43.32. Started from 15.40 version. Even when restarting AndroidUI process aka soft reboot it still uninstalls. Each time it leaves base.apk in data/app/com.google.android.youtube-1. Directories multiply each time, ...youtube-2, ...youtube-3, endlessly I guess, max counter I saw was on 10. The first base.apk leftover is always undeletable, only through recovery it can be done.Tried systemizing/debloating, deleting all related to youtube everywhere, installing vanilla, trying to install on it, etc. Systemizing kinda helped, youtube vanced remained after reboot, but it doesnt see accounts so it's still a failure. From a file manager point of view it still cleared the data/app/com.google.android.youtube leaving only broken base.apk but it couldnt touch the sacred system/priv-app so the app survived the purge. There you have it. Scrubbed all the details I could remember. 
P.s. To be honest it's not even that annoying. Just interesting what is the reason.

**Comment by DanteKG** [comment]
The new root installer is causing more trouble than it's worth (signature verification is good if you're into shady apps but for the purposes of YTVanced the new installer is just making things harder). I have the same problem, the app is reverting to stock after reboot. Is there a way to implement a choice between the old and new root installer in Manager?

**Comment by Einheit-101** [comment]
Same issue on ASUS ROG Phone. I will try non-root now...

**Comment by gghhkm** [comment]
> The new root installer is causing more trouble than it's worth (signature verification is good if you're into shady apps but for the purposes of YTVanced the new installer is just making things harder). I have the same problem, the app is reverting to stock after reboot. Is there a way to implement a choice between the old and new root installer in Manager?

Old method doesn't even work anymore just letting y'all know.

**Comment by ranolfi** [comment]
Please reopen; issue is not fixed.

**Comment by DanteKG** [comment]
> 

> Old method doesn't even work anymore just letting y'all know.

Yea, just like @CurssedCoffin I tried to install latest vanced with SAI and got "signature inconsistent" error... So for the old root installer to work, the .apks themselves need to be changed. What a roller coaster ride... sticking with non-root seems to be the best bet for now

**Comment by omarmarstaoy** [comment]
I have the same problem 
Samsung Note 9 
Android 10 
User Interface 2.5

**Comment by mind-overflow** [comment]
I was having this problem on MIUI 11, however, after updating to MIUI 12 it's now solved. I think this might be related to why you need to disable MIUI optimizations, as newer MIUI versions fixed this kind of stuff.

**Comment by mistermojorizin** [comment]
so i was having this problem with android 8.1 on pixel 2. Was trying the latest versions of the manager and youtube vanced. Today, i noticed there's an update - new manager and new youtube. Does this solve the problem for anyone that was having it before?

**Comment by jd1900** [comment]
I've found the solution, at least for Vanced Music (not tested on Youtube Vanced).

It seems that the manager copies root.apk into /data/app/com.google.android.apps.youtube.music--(replace) as base.apk.
However, it should copy arm64_v8a.apk as base.apk instead.

So just copying arm64_v8a.apk into /data/app/com.google.android.apps.youtube.music--(replace) as base.apk solves the problem.

cp /sdcard/Android/data/com.vanced.manager/files/music/root/arm64_v8a.apk /data/app/com.google.android.apps.youtube.music--(replace)/base.apk

I hope this helps you guys ;)

**Comment by X1nto** [comment]
> **I've found the solution**, at least for Vanced Music (not tested on Youtube Vanced).
> 
> It seems that the manager copies root.apk into /data/app/com.google.android.apps.youtube.music--(replace) as base.apk.
> However, it should copy arm64_v8a.apk as base.apk instead.
> 
> So just copying arm64_v8a.apk into /data/app/com.google.android.apps.youtube.music--(replace) as base.apk solves the problem.
> 
> cp /sdcard/Android/data/com.vanced.manager/files/music/root/arm64_v8a.apk /data/app/com.google.android.apps.youtube.music--(replace)/base.apk
> 
> I hope this helps you guys ;)

If it copies arm64_v8a into that folder you'll get a stock music app

**Comment by jd1900** [comment]
> If it copies arm64_v8a into that folder you'll get a stock music app

Unfortunately you're right... I just saw I still had the app. But I got the stock app.

**Comment by mistermojorizin** [comment]
> > If it copies arm64_v8a into that folder you'll get a stock music app
> 
> Unfortunately you're right... I just saw I still had the app. But I got the stock app.

Youtube music works fine for me, maybe because I didn't have a stock youtube music app on my Oreo system. But I tried youtube vanced with the new installer and it made my youtube disappear completely. I had to restore a TWRP backup just to get the old version vanced back. I don't understand why they changed the installation method when everything was working fine on manager 1.2. I've tried going back to manager 1.2, and it's not able to download the new versions of youtube vanced.

**Comment by DemonSlayer86** [comment]
i think google update their security patch to mess up with some not stock app.

**Comment by DanteKG** [comment]
I installed latest Manager v2.0.1 and the issue seems resolved for me. Vanced persists through reboots. Will give an update if something changes

**Comment by mistermojorizin** [comment]
> I installed latest Manager v2.0.1 and the issue seems resolved for me. Vanced persists through reboots. Will give an update if something changes

For you it was reverting back to stock? For me, it makes youtube disappear altogether. I'll try 2.0.1, it's just a pain because i have to restore a twrp backup to get youtube back.

**Comment by DanteKG** [comment]
From what I understand, in order to install Vanced without signature spoofing it bypasses the issue by making a script and gives it to Magisk which runs it at boot time. So vanced isnt actually installed but the stock YT is patched at every boot. If YT disappears after reboot you can take a look at Magisk logs and better yet a logcat during boot time

**Comment by jd1900** [comment]
> From what I understand, in order to install Vanced without signature spoofing it bypasses the issue by making a script and gives it to Magisk which runs it at boot time. So vanced isnt actually installed but the stock YT is patched at every boot. If YT disappears after reboot you can take a look at Magisk logs and better yet a logcat during boot time

For the phone to do that, AFAIK, Magisk should have a module from Vanced Manager, but I don't see any.

**Comment by DanGLES3** [comment]
> > From what I understand, in order to install Vanced without signature spoofing it bypasses the issue by making a script and gives it to Magisk which runs it at boot time. So vanced isnt actually installed but the stock YT is patched at every boot. If YT disappears after reboot you can take a look at Magisk logs and better yet a logcat during boot time
> 
> For the phone to do that, AFAIK, Magisk should have a module from Vanced Manager, but I don't see any.

A module is not required for a boot script, Mahisk has its own service.d and post-fs-data folders in data/adb, if a root app wants it can simply install its script in there without having to install a whole module

**Comment by X1nto** [comment]
> > > From what I understand, in order to install Vanced without signature spoofing it bypasses the issue by making a script and gives it to Magisk which runs it at boot time. So vanced isnt actually installed but the stock YT is patched at every boot. If YT disappears after reboot you can take a look at Magisk logs and better yet a logcat during boot time
> > 
> > For the phone to do that, AFAIK, Magisk should have a module from Vanced Manager, but I don't see any.
> 
> A module is not required for a boot script, Magisk has its own individual service.d and post-fs-data folders in data/adb, if a root app wants it can simply install its script in there without having to install a whole module 
> 
> Vanced manager installs a Vanced.sh script in service.d which binds the base.apk from Vanced located in data/adb/Vanced into the corresponding stock YouTube apk

This is exactly what it does but unfortunately script doesn't work well in some ROMs, we're trying to fix it but I don't think it'll be fixed anytime soon

**Comment by jd1900** [comment]
> This is exactly what it does but unfortunately script doesn't work well in some ROMs, we're trying to fix it but I don't think it'll be fixed anytime soon

If it is a matter of lack of possibility of testing, just let us know.

**Comment by parcou** [comment]
New to Vance YT under the manager and also trying to research and understand why it goes back to the standard YT after a reboot. I just updated my OP7Pro after year. I am rooted and running Magisk....so I am clear...there is NO fix currently, right...in research still?

What I have done is pulled an older version of YT Vanced Black magisk rep v14.21.54_b01.40.20. I installed the zip in Magisk it works and stays...much older. Side question was that the latest release for repo magisk install? Thx

**Comment by mistermojorizin** [comment]
> > > > From what I understand, in order to install Vanced without signature spoofing it bypasses the issue by making a script and gives it to Magisk which runs it at boot time. So vanced isnt actually installed but the stock YT is patched at every boot. If YT disappears after reboot you can take a look at Magisk logs and better yet a logcat during boot time
> > > 
> > > 
> > > For the phone to do that, AFAIK, Magisk should have a module from Vanced Manager, but I don't see any.
> > 
> > 
> > A module is not required for a boot script, Magisk has its own individual service.d and post-fs-data folders in data/adb, if a root app wants it can simply install its script in there without having to install a whole module
> > Vanced manager installs a Vanced.sh script in service.d which binds the base.apk from Vanced located in data/adb/Vanced into the corresponding stock YouTube apk
> 
> This is exactly what it does but unfortunately script doesn't work well in some ROMs, we're trying to fix it but I don't think it'll be fixed anytime soon

I run an (unrelated) script on boot. I know where the scripts need to go and the permissions they need to stick. I'm looking at the root folder and there's nothing added after using the manager. Could we get a copy of this script? If that's all it is, I bet I could put it in the right place to run at boot.

Also, I'm running the newest version of Magisk (21.1), so I wonder if it has anything to do with that.

**Comment by DanGLES3** [comment]
> > > > > From what I understand, in order to install Vanced without signature spoofing it bypasses the issue by making a script and gives it to Magisk which runs it at boot time. So vanced isnt actually installed but the stock YT is patched at every boot. If YT disappears after reboot you can take a look at Magisk logs and better yet a logcat during boot time
> > > > 
> > > > 
> > > > For the phone to do that, AFAIK, Magisk should have a module from Vanced Manager, but I don't see any.
> > > 
> > > 
> > > A module is not required for a boot script, Magisk has its own individual service.d and post-fs-data folders in data/adb, if a root app wants it can simply install its script in there without having to install a whole module
> > > Vanced manager installs a Vanced.sh script in service.d which binds the base.apk from Vanced located in data/adb/Vanced into the corresponding stock YouTube apk
> > 
> > This is exactly what it does but unfortunately script doesn't work well in some ROMs, we're trying to fix it but I don't think it'll be fixed anytime soon
> 
> I run an (unrelated) script on boot. I know where the scripts need to go and the permissions they need to stick. I'm looking at the root folder and there's nothing added after using the manager. Could we get a copy of this script? If that's all it is, I bet I could put it in the right place to run at boot.
> 
> Also, I'm running the newest version of Magisk (21.1), so I wonder if it has anything to do with that.

The Vanced script is not in the root folder, it's on /data/adb/service.d

**Comment by Toshiba1127** [comment]
Can't we have at least an alternative installer which require disabling signature check like the older version in case the new way wouldn't work? I don't mind leaving signature disabled, and been doing it for years without problem since I use some shady unsigned apks

**Comment by gghhkm** [comment]
> Can't we have at least an alternative installer which require disabling signature check like the older version in case the new way wouldn't work? I don't mind leaving signature disabled, and been doing it for years without problem since I use some shady unsigned apks

No. I doesn't work and probably will never work again. Even SAI doesn't work anymore just probably won't ever work again according to KevinX8.

**Comment by AinzTheSupremeOne** [comment]
Hey there, one user apparently mentioned on Telegram that YouTube Vanced doesn't disappear for them if they disable play protect. Is any of you guys available to test if it works? I am a non root alpha tester 😅. Try disabling Google Play protect and then reinstall Vanced. I am curious to see if it works for you guys

**Comment by gghhkm** [comment]
> Hey there, one user apparently mentioned on Telegram that YouTube Vanced doesn't disappear for them if they disable play protect. Is any of you guys available to test if it works? I am a non root alpha tester 😅. Try disabling Google Play protect and then reinstall Vanced. I am curious to see if it works for you guys

Who ever made that claim is a lying dumbass.
Play Protect don't effect boot scripts in anyway shape or form. 😑

**Comment by X1nto** [comment]
> > Hey there, one user apparently mentioned on Telegram that YouTube Vanced doesn't disappear for them if they disable play protect. Is any of you guys available to test if it works? I am a non root alpha tester 😅. Try disabling Google Play protect and then reinstall Vanced. I am curious to see if it works for you guys
> 
> Who ever made that claim is a lying dumbass.
> Play Protect don't effect boot scripts in anyway shape or form. 😑

Might still give it a try, maybe it's true after all

**Comment by gghhkm** [comment]
> > > Hey there, one user apparently mentioned on Telegram that YouTube Vanced doesn't disappear for them if they disable play protect. Is any of you guys available to test if it works? I am a non root alpha tester 😅. Try disabling Google Play protect and then reinstall Vanced. I am curious to see if it works for you guys
> > 
> > 
> > Who ever made that claim is a lying dumbass.
> > Play Protect don't effect boot scripts in anyway shape or form. 😑
> 
> Might still give it a try, maybe it's true after all

Just did. it didn't change shit, just like I thought.

**Comment by AinzTheSupremeOne** [comment]
> > Hey there, one user apparently mentioned on Telegram that YouTube Vanced doesn't disappear for them if they disable play protect. Is any of you guys available to test if it works? I am a non root alpha tester 😅. Try disabling Google Play protect and then reinstall Vanced. I am curious to see if it works for you guys
> 
> Whoever made that claim is a lying dumbass.
> Play Protect don't effect boot scripts in anyway shape or form. 😑

Figured. But still wanted you guys to give it a try anyways. 😏 And it doesn't work unfortunately as you said 🤭 FYI, it was reported by Milind on the beta testing channel. He said a user on Telegram told him this. Here's the link to the message https://discord.com/channels/328493314485518336/358967484243640321/778232075115429898

**Comment by Toshiba1127** [comment]
> No. It doesn't work and probably will never work again. Even SAI doesn't work anymore and probably won't ever work again according to KevinX8.

Or maybe auto merge downloaded split APK into one and install? I really prefer single APK tho.
Sounds complicated to work with but there is a tool called SAP (Split APKs Packer) by Kirlif' for PC that does the job. There is an app for Android which i don't remember

**Comment by Einheit-101** [comment]
There is no way that this is impossible to resolve with a friggin root access on our end... Wtf.

**Comment by X1nto** [comment]
> There is no way that this is impossible to resolve with a friggin root access on our end... Wtf.

We never said it's impossible

**Comment by FlashOnFire** [comment]
Hi, same problem here.
Before seeing this issue, i already tried to detach app from Play Store (with magisk "Detach" module) and uninstalling stock Youtube app (with magisk "Debloat" module) and it still doesn't work....
As soon as i reboot, the app simply disappear and i need to reinstall it through the Vanced Manager.

**Comment by ranolfi** [comment]
Before commenting "me too" please refer to https://github.com/YTVanced/VancedManager/issues/212#issuecomment-718172997 and use the non-root version until issue is fixed. This is confirmed already and not trivial to solve.

This conversation is growing needlessly, we should not overwhelm the developers. This is how issues get locked.

**Comment by BaIanced** [comment]
Lucky patcher custom patch is working. Using that until a fix is found. Using vanced non root causes duplicate cast destinations due to microg.

**Comment by gghhkm** [comment]
> Using vanced non root causes duplicate cast destinations due to microg.

There's literally a setting in MicroG specifically to fix that. 
![Uploading xvjhrhdhvjgjfj.jpg…]()

**Comment by parcou** [comment]
> Lucky patcher custom patch is working. Using that until a fix is found. Using vanced non root causes duplicate cast destinations due to microg.

What's Lucky patcher for new folks like me?

**Comment by san7cs** [comment]
> > Lucky patcher custom patch is working. Using that until a fix is found. Using vanced non root causes duplicate cast destinations due to microg.
> 
> What's Lucky patcher for new folks like me?

It's an apk made by  Russian dev Chelpus.It's helpful in patching apps/hacking games and several kind of modifications to android system and apps... Overall it's the most powerful small sized root/non root app u ever have wished.

**Comment by san7cs** [comment]
https://t.me/ModByNTH/430

checkout..

**Comment by Justdigit** [comment]
> Lucky patcher custom patch is working. Using that until a fix is found. Using vanced non root causes duplicate cast destinations due to microg.

I don't see a patch in Lucky patcher. They are all old. 
You a patch on app level?

**Comment by BaIanced** [comment]
> > Lucky patcher custom patch is working. Using that until a fix is found. Using vanced non root causes duplicate cast destinations due to microg.
> 
> I don't see a patch in Lucky patcher. They are all old.
> You a patch on app level?

I patched YouTube Music with lucky patcher. Vanced youtube installs fine for me

**Comment by Justdigit** [comment]
> https://t.me/ModByNTH/430
> 
> checkout..

Thanks mate... This works for me. I also happen to have a MI9

**Comment by san7cs** [comment]
> > https://t.me/ModByNTH/430
> > checkout..
> 
> Thanks mate... This works for me. I also happen to have a MI9

https://vancedtube.com/root/download/YouTube_Vanced-v15.05.54-root-default-dark.apks

Not the latest,but not the oldest also.Use SAI to install

**Comment by EnomiZ** [comment]
Same issue here on oneplus 3

**Comment by mingxiaoyu** [comment]
Brand: Mi 9
Operating System: MIUI 12.0.5
Android Version: Android 10
Vanced Manager Version: 2.0.1
YT Vanced: 15.43.32

same issue.

**Comment by twu2** [comment]
I've found the reason for this issue.
in my /data/adb/service.d/vanced.sh
the sleep code like this: 
`while [ "1" != "1" ] ; do sleep 1; done`
but in the source, it should be: 
`while [ "\`getprop sys.boot_completed | tr -d '\r'\ `" != "1" ] ; do sleep 1; done`

the getprop code was executed when generate vanced.sh, so we got wrong script.
after modify vanced.sh, it work fine now.

**Comment by jd1900** [comment]
> I've found the reason for this issue.
> in my /data/adb/service.d/vanced.sh
> the sleep code like this:
> `while [ 1 != 1 ] ; do sleep 1; done`
> but in the source, it should be:
> `` while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 1; done ``
> 
> the getprop code was executed when generate vanced.sh, so we got wrong script.
> after modify vanced.sh, it work fine now.

I've tried with the Music Vanced version (music.sh) and it actually works as well!

Thank you.

**Comment by mingxiaoyu** [comment]
> I've found the reason for this issue.
> in my /data/adb/service.d/vanced.sh
> the sleep code like this:
> `while [ 1 != 1 ] ; do sleep 1; done`
> but in the source, it should be:
> `` while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 1; done ``
> 
> the getprop code was executed when generate vanced.sh, so we got wrong script.
> after modify vanced.sh, it work fine now.

the icon isn't disappear.  but  click it.  the app   can't open

**Comment by ertmuirm** [comment]
> I've found the reason for this issue.
> in my /data/adb/service.d/vanced.sh
> the sleep code like this:
> `while [ 1 != 1 ] ; do sleep 1; done`
> but in the source, it should be:
> `` while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 1; done ``
> 
> the getprop code was executed when generate vanced.sh, so we got wrong script.
> after modify vanced.sh, it work fine now.

this worked for me on vanced music as well, thank you. for vanced youtube, i never had the disappearing app issue, even though the script was the same as vanced music (before manually changing it)

**Comment by EnomiZ** [comment]
I tried to modify vanced.sh, rebooted and nothing has changed, it still disappears

**Comment by BaIanced** [comment]
I can confirm this worked for music.

Those having issues, keep in mind the filenames in /data/adb/service.d
`vanced.sh
music sh`

**Comment by gghhkm** [comment]
> I've found the reason for this issue.
> in my /data/adb/service.d/vanced.sh
> the sleep code like this:
> `while [ 1 != 1 ] ; do sleep 1; done`
> but in the source, it should be:
> `` while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 1; done ``
> 
> the getprop code was executed when generate vanced.sh, so we got wrong script.
> after modify vanced.sh, it work fine now.
Holy shit, I'll tell Xinto now!!

**Comment by twu2** [comment]
First, make sure you write a correct delay code in script.
If it still not work, try to add more delay before mount command.

Vanced use mount bind to replace regular app's apk.
But if we do this too fast (maybe before PacakgeManager check the apk), it will reject to load suck apk file (that's why we don't replace it directly).

**Comment by pellaaa93** [comment]
> First, make sure you write a correct delay code in script.
> If it still not work, try to add more delay before mount command.
> 
> Vanced use mount bind to replace regular app's apk.
> But if we do this too fast (maybe before PacakgeManager check the apk), it will reject to load such apk file (that's why we don't replace it directly).

what do i have to write to add delay before mount command? thanks

**Comment by skyleet** [comment]
> I've found the reason for this issue.
> in my /data/adb/service.d/vanced.sh
> the sleep code like this:
> `while [ 1 != 1 ] ; do sleep 1; done`
> but in the source, it should be:
> `` while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 1; done ``
> 
> the getprop code was executed when generate vanced.sh, so we got wrong script.
> after modify vanced.sh, it work fine now.

Thanks man. It fixed the issue for me.

**Comment by ralphocdol** [comment]
> I've found the reason for this issue.
> in my /data/adb/service.d/vanced.sh
> the sleep code like this: 
> `while [ 1 != 1 ] ; do sleep 1; done`
> but in the source, it should be: 
> ```while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 1; done```
> 
> the getprop code was executed when generate vanced.sh, so we got wrong script.
> after modify vanced.sh, it work fine now.

Works for me too for music.sh, though I did not bother changing vanced.sh since it's already working fine.

**Comment by skyleet** [comment]
Solution provided by @twu2  working perfectly for both YouTube and yt music. Devs plz fix the mount script.

My OS:
Device: Redmi note 4 mido
Android: Oreo 8.1
OS: lineage OS

**Comment by KrishnaMoorthy12** [comment]
> Solution provided by @twu2 working perfectly for both YouTube and yt music. Devs plz fix the mount script.
> 
> My OS:
> Device: Redmi note 4 mido
> Android: Oreo 8.1
> OS: lineage OS

The fix is made in source. We can expect a release soon...!
Commit: 59193c528c16a30c4d63b522d876d09f4329bc3c by @KoopahTManiac

**Comment by Hinfty** [comment]
> I've found the reason for this issue.
> in my /data/adb/service.d/vanced.sh
> the sleep code like this:
> `while [ 1 != 1 ] ; do sleep 1; done`
> but in the source, it should be:
> `` while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 1; done ``
> 
> the getprop code was executed when generate vanced.sh, so we got wrong script.
> after modify vanced.sh, it work fine now.

This workaround does not fix it for me. 
Yt music uninstalled/disappeared for me every reboot, Vanced was fine. 
With the fix, it does not disappear, but is replaced by the stock Yt music.

**Comment by gghhkm** [comment]
> > First, make sure you write a correct delay code in script.
> > If it still not work, try to add more delay before mount command.
> > Vanced use mount bind to replace regular app's apk.
> > But if we do this too fast (maybe before PacakgeManager check the apk), it will reject to load such apk file (that's why we don't replace it directly).
> 
> what do i have to write to add delay before mount command? thanks

I believe you change the number after "sleep"

Sleep 1 = 1s,, Sleep 2 = 2s.

So maybe try :
while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 2; done

**Comment by saikatm** [comment]
> > > First, make sure you write a correct delay code in script.
> > > If it still not work, try to add more delay before mount command.
> > > Vanced use mount bind to replace regular app's apk.
> > > But if we do this too fast (maybe before PacakgeManager check the apk), it will reject to load such apk file (that's why we don't replace it directly).
> > 
> > 
> > what do i have to write to add delay before mount command? thanks
> 
> I believe you change the number after "sleep"
> 
> Sleep 1 = 1s,, Sleep 2 = 2s.
> 
> So maybe try :
> while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 2; done

thanks alot man this worked for me for the vanced music app was looking for a fix from few days

**Comment by SOUMIK124** [comment]
I also have that same problem
Device: Samsung galaxy j200h
Android version:5.1.1(lollipop)
Vanced manager version:2.0.1

**Comment by rjshrjndrn** [comment]
Till the official release happens, you can use the debug build(stripped out from the github actions) to install the vanced youtube and music.
> This app is debug build and not production ready. So once you installed the apps, uninstall the manager. 

[link](https://github.com/YTVanced/VancedManager/suites/1624562700/artifacts/29887549)

**Comment by X1nto** [comment]
> Till the official release happens, you can use the debug build(stripped out from the github actions) to install the vanced youtube and music.
> > This app is debug build and not production ready. So once you installed the apps, uninstall the manager. 
> 
> [link](https://github.com/YTVanced/VancedManager/suites/1624562700/artifacts/29887549)

Keep in mind that debug apk maybe buggy and will definitely lag on some devices

**Comment by soredake** [comment]
Tried this apk, nothing changed, youtube vanished after reboot.

**Comment by Amanse** [comment]
> I've found the reason for this issue.
> in my /data/adb/service.d/vanced.sh
> the sleep code like this:
> `while [ 1 != 1 ] ; do sleep 1; done`
> but in the source, it should be:
> `` while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 1; done ``
> 
> the getprop code was executed when generate vanced.sh, so we got wrong script.
> after modify vanced.sh, it work fine now.

OUR SAVIOUR!

**Comment by maahmoudsamir** [comment]
> I've found the reason for this issue.
> in my /data/adb/service.d/vanced.sh
> the sleep code like this:
> `while [ 1 != 1 ] ; do sleep 1; done`
> but in the source, it should be:
> `` while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 1; done ``
> 
> the getprop code was executed when generate vanced.sh, so we got wrong script.
> after modify vanced.sh, it work fine now.

Not working in vanced.sh ( youtube vanced )
And try more 
Phone : note 8 pro
Miui 11 android 9

**Comment by BearTS** [comment]
> I've found the reason for this issue.
> in my /data/adb/service.d/vanced.sh
> the sleep code like this: 
> `while [ 1 != 1 ] ; do sleep 1; done`
> but in the source, it should be: 
> ```while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 1; done```
> 
> the getprop code was executed when generate vanced.sh, so we got wrong script.
> after modify vanced.sh, it work fine now.

Worked like a charm

**Comment by rjshrjndrn** [comment]
> Tried this apk, nothing changed, youtube vanished after reboot.

Hmmm that's weird, for me the script didn't work. But this apk worked. 

Steps:
Uninstalled original manager
Installed this apk
Rebooted phone to make sure vanced(in my case YT Music)was missing
Installed YT Music from this apk
Rebooted the phone.
Uninstalled the apk

**Comment by c7r4x** [comment]
![Screenshot_2020-12-10-20-56-03-463_com miui bugreport](https://user-images.githubusercontent.com/74525590/101823385-f4672e00-3b2a-11eb-89a3-210ef7dced9c.jpg)
Still Uninstalling after reboot #

**Comment by darkdex52** [comment]
> > > First, make sure you write a correct delay code in script.
> > > If it still not work, try to add more delay before mount command.
> > > Vanced use mount bind to replace regular app's apk.
> > > But if we do this too fast (maybe before PacakgeManager check the apk), it will reject to load such apk file (that's why we don't replace it directly).
> > 
> > 
> > what do i have to write to add delay before mount command? thanks
> 
> I believe you change the number after "sleep"
> 
> Sleep 1 = 1s,, Sleep 2 = 2s.
> 
> So maybe try :
> while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 2; done

This still didn't fix it for me. Youtube Vanced disappears after every reboot, but normal Youtube isn't back, it crashes on startup.

Phone: Samsung S20 FE

**Comment by twu2** [comment]
> This still didn't fix it for me. Youtube Vanced disappears after every reboot, but normal Youtube isn't back, it crashes on startup.

you can try this:
1. install Youtube Vanced from Manager.
2. move /data/adb/service.d/vanced.sh to /data/adb/vanced.sh
3. reboot
4. after reboot, the Youtube should be normal one.
5. execute command: su -Mc /data/adb/vanced.sh
6. remove Youtube from recent task, and open it again, it should be Youtube Vanced now.

if this work, add more sleep between while and mount should work (not the sleep between do and done).

**Comment by King-Che** [comment]
> I've found the reason for this issue.
> in my /data/adb/service.d/vanced.sh
> the sleep code like this:
> `while [ 1 != 1 ] ; do sleep 1; done`
> but in the source, it should be:
> `` while [ "`getprop sys.boot_completed | tr -d '\r'`" != "1" ] ; do sleep 1; done ``
> 
> the getprop code was executed when generate vanced.sh, so we got wrong script.
> after modify vanced.sh, it work fine now.

This worked for me lile charm, thanks alot ❤️

**Comment by ObscenityIB** [comment]
I see people saying it was a script doing it.

My theory was Google Play was uninstalling it, because i noticed when vanced is installed, installing ANYTHING, even a small 300KB app, takes a really long time

But installs fine if vanced is not installed.

**Comment by gghhkm** [comment]
> I see people saying it was a script doing it.
> 
> My theory was Google Play was uninstalling it, because i noticed when vanced is installed, installing ANYTHING, even a small 300KB app, takes a really long time
> 
> But installs fine if vanced is not installed.

It's not caused by Google play.
Google play only causes it to revert to Stock which is because it Auto updates, which is easily fixed with the "Detach" magisk module.

**Comment by Hinfty** [comment]
> It's not caused by Google play.
> Google play only causes it to revert to Stock which is because it Auto updates, which is easily fixed with the "Detach" magisk module.

Detach never worked 100% for me, it would just update anyway sometimes.
But now we can deactivate auto update on an app-to-app basis and that's fantastic.

**Comment by soredake** [comment]
>  But now we can deactivate auto update on an app-to-app basis and that's fantastic.

Google play still updated youtube many times even when auto-updates disabled for app.

**Comment by ObscenityIB** [comment]
> Google play only causes it to revert to Stock which is because it Auto updates

even when auto update is not enabled?

**Comment by ObscenityIB** [comment]
> Google play still updated youtube many times even when auto-updates disabled for app.

oh, well thats dumb

**Comment by Hinfty** [comment]
> > But now we can deactivate auto update on an app-to-app basis and that's fantastic.
> 
> Google play still updated youtube many times even when auto-updates disabled for app.

Actually that just sounds like a bug in the Play Store.

**Comment by X1nto** [comment]
Disappearing issue should be fixed in the new installer, just reinstall vanced/music and that's it. Regarding the crash issue after rebooting, it's yet unknown why that happens, we're still investigating that issue. This thread became hard to follow because of large amount of comments, if you have other related issue, please open a new issue and avoid commenting "I have the same issue" without posting some basic device info. I'll lock and close this thread now
